pandas/core/dtypes/cast.py

```diff
@@ -1018,6 +1019,80 @@ def soft_convert_objects(
     return values
 
 
+def convert_dtypes(
+    input_array,
+    convert_string: bool = True,
+    convert_integer: bool = True,
+    convert_boolean: bool = True,
+):
+    """
+        Convert objects to best possible type, and optionally,
+        to types supporting ``pd.NA``.
+
+        Parameters
+        ----------
+        input_array : ExtensionArray or PandasArray
+        convert_string : bool, default True
+            Whether object dtypes should be converted to ``StringDtype()``.
+        convert_integer : bool, default True
+            Whether, if possible, conversion can be done to integer extension types.
+        convert_boolean : bool, defaults True
+            Whether object dtypes should be converted to ``BooleanDtypes()``.
+
+        Returns
+        -------
+        Array
```

Comment:
This function returns a dtype, not an array?

Reply:
fixed